### PR TITLE
Add read_parquet_metadata to pylibcudf

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
Contributes to https://github.com/rapidsai/cudf/issues/15162

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/pylibcudf/pylibcudf/io/parquet_metadata.pyx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        return ParquetColumnSchema.from_column_schema(self.column_schema.child(idx))
+
+    cpdef list children(self):
+        """
+        Returns schemas of all child columns.
+
+        Returns
+        -------
+        list[ParquetColumnSchema]
+            Children schemas.
```
Comment (wence-) [inline python/pylibcudf/pylibcudf/io/parquet_metadata.pyx:71]:
```suggestion
            Child schemas.
```
Or:
```suggestion
            Schemas of children.
```

File: python/pylibcudf/pylibcudf/io/parquet_metadata.pyx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        Returns
+        -------
+        int
+            Number of row groups.
+        """
+        return self.meta.num_rowgroups()
+
+    cpdef dict metadata(self):
+        """
+        Returns the Key value metadata in the file footer.
```
Comment (wence-) [inline python/pylibcudf/pylibcudf/io/parquet_metadata.pyx:163]:
```suggestion
        Returns the key-value metadata in the file footer.
```

File: python/pylibcudf/pylibcudf/io/parquet_metadata.pyx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        return self.meta.metadata()
+
+    cpdef list rowgroup_metadata(self):
+        """
+        Returns the row group metadata in the file footer.
+
+        Returns
+        -------
+        list[dict[str, int]]
+            Vector of row group metadata as maps.
```
Comment (wence-) [inline python/pylibcudf/pylibcudf/io/parquet_metadata.pyx:179]: *nit*: why are we decoding the rowgroup metadata but not the metadata keys?

**Review by wence- — APPROVED** [review]
Tiny quibbles, nothing blocking.

File: python/pylibcudf/pylibcudf/io/parquet_metadata.pyx (hunk 3 above)
Comment (mroeschke) [reply python/pylibcudf/pylibcudf/io/parquet_metadata.pyx:179]: That's fair. To be more Python-like I'll also decide the metadata keys

**Pull request merged**